File: src/main/java/com/camackenzie/exvi/core/AccountAccessKeyResult.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core;

/**
 *
 * @author callum
 */
public class AccountAccessKeyResult {

    private final int error;
    private final String message;
    private final String accessKey;

    public AccountAccessKeyResult(int error,
            String message,
            String accessKey) {
        this.error = error;
        this.message = message;
        this.accessKey = accessKey;
    }

    public int getError() {
        return this.error;
    }

    public String getMessage() {
        return this.message;
    }

    public String getAccessKey() {
        return this.accessKey;
    }

}

File: common/src/commonMain/kotlin/com/camackenzie/exvi/core/api/AccountAccessKeyResult.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core.api

import com.camackenzie.exvi.core.util.EncodedStringCache
import com.camackenzie.exvi.core.util.SelfSerializable
import com.camackenzie.exvi.core.util.cached
import kotlinx.serialization.json.*
import kotlinx.serialization.*


/**
 *
 * @author callum
 */
@kotlinx.serialization.Serializable
class AccountAccessKeyResult : DataResult<EncodedStringCache>, SelfSerializable {
    override val result: EncodedStringCache?

    constructor(
        error: Int,
        message: String,
        accessKey: String
    ) : super(error, message) {
        result = accessKey.cached()
    }

    constructor(msg: String, key: String) : this(0, msg, key) {}
    constructor(err: Int, msg: String) : this(err, msg, "") {}

    val accessKey: String
        get() = this.result!!.get()

    override fun toJson(): String {
        return Json.encodeToString(this)
    }

    override fun getUID(): String {
        return "AccountAccessKeyResult"
    }
}